common/src/main/java/io/druid/collections/BlockingPool.java

```diff
@@ -77,15 +77,16 @@ public int getPoolSize()
    * Take a resource from the pool, waiting up to the
    * specified wait time if necessary for an element to become available.
    *
-   * @param timeout maximum time to wait for a resource, in milliseconds.
+   * @param timeoutMs maximum time to wait for a resource, in milliseconds.
    *
    * @return a resource, or null if the timeout was reached
    */
-  public ReferenceCountingResourceHolder<T> take(final long timeout)
+  public ReferenceCountingResourceHolder<T> take(final long timeoutMs)
```

Comment:
It still has non-obvious undocumented corner-case behaviour if timeoutMs=0, than it waits indefinitely, rather than not waiting at all.